vars/prepareDefaultValues.groovy

```diff
@@ -1,10 +1,26 @@
 import com.sap.piper.DefaultValueCache
 
+import hudson.AbortException
+
 def call(Map parameters = [:]) {
     handlePipelineStepErrors (stepName: 'prepareDefaultValues', stepParameters: parameters) {
         if(!DefaultValueCache.getInstance()) {
             Map defaultValues  = readYaml text: libraryResource('default_pipeline_environment.yml')
-            DefaultValueCache.createInstance(defaultValues)
+            Map customDefaultValues = null
+
+            def customDefaults = null
+
+            try {
+              customDefaults = libraryResource('pipeline_environment.yml')
```

Comment:
we should be able to support more than one custom default file (ordered List) and allow custom names.

Reply:
@CCFenner already suggested this when we talked about this feature. IMO one additional config file should be sufficient in order to describe a system landscape. We shouldn't make it more complex unless there is a real use case justifying this. @OliverNocon @CCFenner : which use case for several config files do you have in mind?

Technically it is not an issue to have something like customerDefault-$i where $i is a number, would also be possible to define addtional config files in earlier config files ...

Reply:
For me 2 custom configs come in very easily. 
Just consider SAP overall and then some bigger area within SAP ...

I suggest that we consider this right from the start.

Reply:
company/department scenarios could also be handled by merging company and department settings into one additional config file and incorporate the pre-merged file. E.g. in maven it is also not possible to have several ```settings.xml```'s.

We get problems which we do not have with only one "landscape" file. E.g. what happens when we have a setup where we have name clashes? e.g.  big company file is: ```pipeline_environment-0.yml``` and somewhere lower in the hierarchy we get clashes like having ```pipeline_environment-5.yml``` twice provided by two different departments. In case we solve it like this: ```seq 1 n``` : what is a suitable value for ```n```?

How should we configure the several files? Would be possible to use an enviornment variables holding a comma separated list, something like ```pipeline_environment-company.yml,pipeline_environment-dep-1.yml,pipeline_environment-dep-2.yml```. How should be act in this case if one of the config files is not found.

All this kind of problems we can avoid if to do not provide additional layers. Again: I'm not fully against several layers, but it adds complexity. This needs to be justified. I personally prefer to start simple (I' a fried of the KISS principle anyway). Wait until we get feedback like "it is not sufficient to have only layer" and react afterwards.

Reply:
I would agree with @OliverNocon to have this right in the first place. I think is would be a bit cumbersome to request the users to merge there custom config in a companies default config. 

However we could start simple with one file, but I would like a avoid migration effort in the end. So we should come around with a solution that could be extended easily.